pywhy_stats/independence.py

```diff
@@ -44,5 +53,30 @@ def independence_test(
     result : PValueResult
         An instance of the PValueResult data class, containing the p-value, test statistic,
         and any additional information related to the independence test.
+
+    See Also
+    --------
+    fisherz : Fisher's Z test for independence
     """
-    pass
+    method_func: ModuleType
+    if method == Methods.AUTO:
+        method_func = Methods.FISHERZ
+    else:
+        method_func = method
+
+    if method_func == Methods.FISHERZ:
+        if condition_on is None:
+            data = [X, Y]
+        else:
+            data = [X, Y, condition_on]
+        for _data in data:
+            _, pval = scipy.stats.normaltest(_data)
+
+            # XXX: we should add pinguoin as an optional dependency for doing multi-comp stuff
+            if pval < 0.05:
+                warn("Your data is not all normal but you're trying to use partial correlation")
```

Comment:
The wording "your data is not all normal" sounds a bit confusing. What about "The provided data does not seem to be Gaussian, but the Fisher-Z test assumes that the data follows a Gaussian distribution. The result should be interpreted carefully or consider a different independence test method." or something along the line.